internal/services/cognitive/azure_ai_services_resource.go

```diff
@@ -0,0 +1,757 @@
+
+			state := AzureAIServicesResourceResourceModel{}
+			id, err := cognitiveservicesaccounts.ParseAccountID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.AccountsGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			keys, err := client.AccountsListKeys(ctx, *id)
+			if err != nil {
+				// note for the resource we shouldn't gracefully fail since we have permission to CRUD it
+				return fmt.Errorf("listing the Keys for %s: %+v", id, err)
+			}
+
+			if model := keys.Model; model != nil {
+				state.PrimaryAccessKey = pointer.From(model.Key1)
+				state.SecondaryAccessKey = pointer.From(model.Key2)
+			}
+
+			state.Name = id.AccountName
+			state.ResourceGroupName = id.ResourceGroupName
+
+			if model := resp.Model; model != nil {
+				state.Location = location.NormalizeNilable(model.Location)
+				if sku := model.Sku; sku != nil {
+					state.SkuName = sku.Name
+				}
+
+				identityFlatten, err := identity.FlattenSystemAndUserAssignedMapToModel(model.Identity)
+				if err != nil {
+					return err
+				}
+				state.Identity = *identityFlatten
+
+				if props := model.Properties; props != nil {
+					state.Endpoint = pointer.From(props.Endpoint)
+					state.CustomSubdomainName = pointer.From(props.CustomSubDomainName)
+					state.NetworkACLs = flattenAzureAIServicesNetworkACLs(props.NetworkAcls)
+					state.Fqdns = pointer.From(props.AllowedFqdnList)
+
+					state.PublicNetworkAccess = string(*props.PublicNetworkAccess)
+
+					outboundNetworkAccessRestricted := false
+					if props.RestrictOutboundNetworkAccess != nil {
+						outboundNetworkAccessRestricted = *props.RestrictOutboundNetworkAccess
+					}
+					state.OutboundNetworkAccessRestricted = outboundNetworkAccessRestricted
+
+					localAuthEnabled := true
+					if props.DisableLocalAuth != nil {
+						localAuthEnabled = !*props.DisableLocalAuth
+					}
+					state.LocalAuthorizationEnabled = localAuthEnabled
+
+					customerManagedKey, err := flattenAzureAIServicesCustomerManagedKey(props.Encryption, env)
+					if err != nil {
+						return err
+					}
+					state.CustomerManagedKey = customerManagedKey
+				}
+
+				state.Tags = pointer.From(model.Tags)
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (AzureAIServicesResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.AccountsClient
+
+			var model AzureAIServicesResourceResourceModel
+
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			id, err := cognitiveservicesaccounts.ParseAccountID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf(" Cannot parse Azure AI service ID: %s", err)
```

Comment:
We won't wrap error messages returned by ID parsing functions since these already all the relevant information
```suggestion
				return err
```